tests/test_imgurfetcher.py

```diff
@@ -108,25 +181,87 @@ def test_select_image(self):
+
+        self.fetcher.min_width = 1000
+        self.fetcher.min_height = 1000
+        # trigger rejecting bc of width
+        result = self.fetcher._select_image([self.bad_image_width,
+                                             self.good_image])
+        self.assertEquals(result, self.good_image)
+
+        # trigger rejecting bc of height
+        result = self.fetcher._select_image([self.bad_image_height,
+                                             self.good_image])
+        self.assertEquals(result, self.good_image)
+
+        # trigger rejecting bc of title
+        result = self.fetcher._select_image([self.bad_image_title,
+                                             self.good_image])
+        self.assertEquals(result, self.good_image)
+
+        # trigger rejecting bc of description
+        result = self.fetcher._select_image([self.bad_image_description,
+                                             self.good_image])
+        self.assertEquals(result, self.good_image)
+
         with patch("bg_daemon.fetchers.imgurfetcher.ImgurClient") as \
                 mock_class:
 
             mock_method = mock_class.return_value.get_album_images
-            mock_method.return_value = [self.gallery[-1]]
+            mock_method.return_value = [self.good_image]
 
             result = self.fetcher._select_image([self.album])
 
             mock_method.assert_called_once_with(self.album.id)
-            self.assertTrue(result == self.gallery[-1])
+            self.assertEquals(result, self.good_image)
+
+            # the album returns non-working images... it should seek more, but
+            # that's it.
+            mock_method.return_value = []
+            result = self.fetcher._select_image([self.album])
+            self.assertTrue(result is None)
+
+            # now try with an image right after it, we should select the last
+            # image
+            result = self.fetcher._select_image([self.album, self.good_image])
+            self.assertEquals(result, self.good_image)
+
+            # finally, imagine that the get_album_image method breaks and
+            # returns none
+            mock_method.return_value = None
+            result = self.fetcher._select_image([self.album, self.good_image])
+            self.assertEquals(result, self.good_image)
+
+        # Test for keyword mode
+        self.fetcher.mode = "keywords"
+        result = self.fetcher._select_image([self.good_image])
+        self.assertEquals(result, self.good_image)
+
+        # test that the fetcher gives up on 30 attempts
+        result = self.fetcher._select_image([self.bad_image_title]*50)
+        self.assertTrue(result is None)
+
+        # return everything to normal
+        self.fetcher.mode = "recent"
 
-    """
-    Tests for input sanity and proper output on the galleryAlbum helper.
-    """
     def test_get_image_from_album(self):
+        """
+
+            Tests for input sanity and proper output on the galleryAlbum
```

Comment:
Minor, but this docstring seems to be over-indented and the line above can be removed.